docs/user_guide/affinity.md

```diff
@@ -0,0 +1,73 @@
+# Affinity
+
+There are times when you want to express a preference (or anti-preference) for
+two instances sharing a hypervisor. For example, you might have two IO intensive
+instances and want to keep them apart, or you might have two instances which
+talk a lot over the network to each other and want to keep them together.
+
+Shaken Fist supports a simple affinity system for this use case. The system is
+built on top of an instance metadata entry called `tags`, where you specify the
+attributes of the instance. This metadata entry must be formatted as a valid
+JSON list. So for example we might set the following simple tag on an instance:
+
+```
+["webserver"]
+```
+
+Or on the command line:
+
+`sf-client instance set-metadata ...uuid... tags '["webserver"]'`
+
+Affinity is then set via the `affinity` metadata entry, and consists of a JSON
+dictionary specifying the weight to give to a given tag in scheduling decisions.
+In our current example, let's say we want to try quite hard to not have more than
+one of our web servers on a given hypervisor. We might therefore write an
+affinity metadata entry like this:
+
+```
+{
+    "webserver": -10
+}
+```
+
+Or on the command line:
+
+`sf-client instance set-metadata ...uuid... affinity '{"webserver": -10}'`
+
+There are no fixed rules for what the maximum and minimum numbers for this
+preference value are, although generally we recommend they range from -100 to 100,
+where 100 means you'd really really love to be on the same hypervisor, and -100
+means you'd be very unhappy to be on the same hypervisor.
+
+Under the hood, Shaken Fist uses the affinity preferences to "tweak" the measured
+CPU load of a given hypervisor to make it more likely to be selected during the
+scheduling decision for an instance. This decision is of course only made on the
+original start up of an instance, and does not apply later. That is, if you change
```

Comment:
I would call out this out in a separate heading/note-box thingy. It could be confusing.

Reply:
Oh, an admonition!